baselines/constraint/constraint.py

```diff
@@ -8,6 +8,42 @@
 
 
 class Constraint(DFA):
+    '''
+    Constraint represents a given constraint to place on the agent.
+       
+    Attributes
+    ==========
+    name : str
+        The name of the constraint
+    reg_ex : str?
+        A representation of the constraint
+        using regular expressions.
+    violation_reward : int
+        The reward signal for violating the
+        constraint. Used in reward shaping.
+    s_tl : ?
+        Unknown. Likely stands for the state
+        translation layer. What that means?
+    a_tl : ?
+        Unknown. Likely stands for the action
+        translation layer. What that means?
+    s_active : boolean
+        A flag indicating if the state 
+        translation layer is in effect?
+    a_active : boolean
+        A flag indicating if the action
+        translation laer is in effect?
+
+    Methods
+    =======
+    __init__(self, name, reg_ex,
+             violation_reward, s_tl,
+             a_tl, s_active, a_active)
+            Initializes a Constraint.
+    step(self, obs,
```

Comment:
Runs one step of the constraint mechanism which takes steps in parallel with the MDP it's attached to